functional-tests/mining_sync_test.go

```diff
@@ -74,53 +77,63 @@ func TestBootstrapMineOnce(t *testing.T) {
+
+	seed := node.MakeChainSeed(t, genCfg)
+	chainClock := clock.NewChainClockFromClock(uint64(genTime), blockTime, fakeClock)
+
+	miner := makeNode(ctx, t, seed, chainClock)
+	_, _, err := initNodeGenesisMiner(t, miner, seed, genCfg.Miners[0].Owner, presealPath, genCfg.Miners[0].SectorSize)
 	require.NoError(t, err)
 
-	node0 := tns[0]
+	err = miner.Start(ctx)
+	require.NoError(t, err)
 
-	// Setup first node, note: Testbed.Name() is the directory
-	genConfigPath := filepath.Join(root, "fixtures/setup.json")
-	genesisConfig := iptbtester.ReadSetup(t, genConfigPath)
+	err = miner.StorageMining.Start(ctx)
+	require.NoError(t, err)
 
-	// set proving period start to something soon
-	start := abi.ChainEpoch(97408490)
-	genesisConfig.Miners[0].ProvingPeriodStart = &start
+	maddr, err := miner.BlockMining.BlockMiningAPI.MinerAddress()
+	require.NoError(t, err)
 
-	genesis := iptbtester.RequireGenesis(t, node0.Testbed.Name(), genesisConfig)
-	genesis.SectorsDir = filepath.Join(node0.Dir(), "sectors")
-	genesis.PresealedSectorDir = filepath.Join(root, "./fixtures/genesis-sectors")
+	status, err := miner.PorcelainAPI.MinerGetStatus(ctx, maddr, requireChainHead(t, miner))
+	require.NoError(t, err)
 
-	node0.MustInitWithGenesis(ctx, genesis)
-	node0.MustStart(ctx)
-	defer node0.MustStop(ctx)
+	require.Equal(t, abi.ChainEpoch(1), status.ProvingPeriodStart)
 
-	var minerAddress address.Address
-	node0.MustRunCmdJSON(ctx, &minerAddress, "go-filecoin", "config", "mining.minerAddress")
+	// mine once to enter proving period
+	fakeClock.Advance(blockTime)
+	_, err = miner.BlockMining.BlockMiningAPI.MiningOnce(ctx)
+	require.NoError(t, err)
 
-	// expect proving period start to be 1 and no failures
-	var status porcelain.MinerStatus
-	node0.MustRunCmdJSON(ctx, &status, "go-filecoin", "miner", "status", minerAddress.String())
-	require.Equal(t, abi.ChainEpoch(97408490), status.ProvingPeriodStart)
-	require.Equal(t, 0, status.PoStFailureCount)
-
-	// start mining
-	node0.MustRunCmd(ctx, "go-filecoin", "mining", "start")
-
-	// Wait for either a changes the proving period start showing a successful PoSt or we see a failure on chain
-	for i := 0; i < 30; i++ {
-		node0.MustRunCmdJSON(ctx, &status, "go-filecoin", "miner", "status", minerAddress.String())
-		require.Equal(t, 0, status.PoStFailureCount, "No PoSt received in proving period window")
-		if status.ProvingPeriodStart > 97408490 {
-			t.Log("Successfully advanced proving period start without incrementing failure count")
+	// Post should have been triggered, simulate mining while waiting for update to proving period start
+	for i := 0; i < 25; i++ {
+		fakeClock.Advance(blockTime)
+		_, err := miner.BlockMining.BlockMiningAPI.MiningOnce(ctx)
+		require.NoError(t, err)
+
+		status, err := miner.PorcelainAPI.MinerGetStatus(ctx, maddr, requireChainHead(t, miner))
+		require.NoError(t, err)
+
+		if status.ProvingPeriodStart > 1 {
 			return
 		}
-		time.Sleep(5 * time.Second)
+
+		time.Sleep(2 * time.Second)
```

Comment:
Do we still really need a non-trivial sleep here? Please jot a comment about what it's waiting for (the PoSt computation?)